sdk/eventhub/Azure.Messaging.EventHubs/src/Core/Guard.cs

```diff
@@ -3,87 +3,31 @@
-            }
-        }
-
-        /// <summary>
-        ///   Ensures that an argument's value is not <c>null</c> or an empty string, throwing an
-        ///   <see cref="ArgumentException" /> if that invariant is not met.
-        /// </summary>
-        ///
-        /// <param name="argumentName">The name of the argument being considered.</param>
-        /// <param name="argumentValue">The value of the argument to verify.</param>
-        ///
-        public static void ArgumentNotNullOrEmpty(string argumentName,
-                                                  string argumentValue)
-        {
-            if (String.IsNullOrEmpty(argumentValue))
-            {
-                throw new ArgumentException(String.Format(CultureInfo.CurrentCulture, Resources.ArgumentNullOrEmpty, argumentName), argumentName);
-            }
-        }
-
-        /// <summary>
-        ///   Ensures that an argument's value is not <c>null</c> or a string comprised of only whitespace,
-        ///   throwing an <see cref="ArgumentException" /> if that invariant is not met.
-        /// </summary>
-        ///
-        /// <param name="argumentName">The name of the argument being considered.</param>
-        /// <param name="argumentValue">The value of the argument to verify.</param>
-        ///
-        public static void ArgumentNotNullOrWhitespace(string argumentName,
-                                                       string argumentValue)
-        {
-            if (String.IsNullOrWhiteSpace(argumentValue))
-            {
-                throw new ArgumentException(String.Format(CultureInfo.CurrentCulture, Resources.ArgumentNullOrWhiteSpace, argumentName), argumentName);
-            }
-        }
-
         /// <summary>
         ///   Ensures that an argument's value is a string comprised of only whitespace, though
         ///   <c>null</c> is considered a valid value.  An <see cref="ArgumentException" /> is thrown
         ///   if that invariant is not met.
         /// </summary>
         ///
-        /// <param name="argumentName">The name of the argument being considered.</param>
         /// <param name="argumentValue">The value of the argument to verify.</param>
-        ///
-        public static void ArgumentNotEmptyOrWhitespace(string argumentName,
-                                                        string argumentValue)
+        /// <param name="argumentName">The name of the argument being considered.</param>
+        /// <exception cref="ArgumentException">The argument is empty or contains only white-space.</exception>
+        public static void NotEmptyOrWhiteSpace(string argumentValue, string argumentName)
         {
-            if (argumentValue == null)
+            if (argumentValue is null)
             {
                 return;
             }
 
-            if (String.IsNullOrWhiteSpace(argumentValue))
+            if (string.IsNullOrWhiteSpace(argumentValue))
```

Comment:
For internal consistency with other uses within Event Hubs, I'd prefer that we leave `String`, rather than changing case for these.

Reply:
Our code style guide mandates `string`

Reply:
News to me; last that I had spoken to Alex about it we didn't have a formalized style guide.   Can you throw me a pointer to it?

Reply:
https://github.com/Azure/azure-sdk-for-net/pull/5528#issuecomment-478807105

Reply:
Fair enough; that was about the time of the last conversation that was had.   Not coming from CoreFx and not being overly familiar with their standards, the nuance was missed in the push for hitting release deadlines.   That'll be sorted out in Dozie's work stream with the `.editorconfig` to make it obvious going forward.

For now, I appreciate and agree with Heath's suggestion to keep things internally consistent and let the formatting pass do everything at once.